core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -41,6 +48,30 @@ class SparkException(
   override def getSqlState: String = SparkThrowableHelper.getSqlState(errorClass.orNull)
 }
 
+class SparkRuntimeException(
+    message: String,
+    cause: Throwable,
```

Comment:
@karenfeng I'd rather define `cause` as type of `Exception`, What do you think?

Reply:
`cause` is always a type of `Throwable` in the base type and it needs to be propagated, so I think we can leave it as is. But I think we can simplify this class to only accept `errorClass` and `messageParameters` (and maybe `cause` if needed); see `SparkArithmeticException` as an example.

Reply:
ok，I will simplify this class to only accept errorClass and messageParameters，also with others. The cause is needed for it in QueryExecutionErrors. Thank you.